### ISSUE TITLE
Cannot provide custom RetryTemplate

### ISSUE THREAD

**Issue by iNikem** [spring-cloud/spring-cloud-commons#283]
Before starting using Spring Cloud Stream our Spring Boot application has custom `RetryTemplate` bean configured for inner needs. When we added Spring Cloud Stream to the application `org.springframework.cloud.client.loadbalancer.LoadBalancerAutoConfiguration.RetryAutoConfiguration` kicked in and has replaced our `RetryTemplate` bean with its own. And with no way to supply our own bean.

**Comment by holy12345** [comment]
@iNikem 
Hello, just read your question, I checked the source code, the source code reads as follows:
```
	@Configuration
	@ConditionalOnClass(RetryTemplate.class)
	public static class RetryAutoConfiguration {
		@Bean
		public RetryTemplate retryTemplate() {
			RetryTemplate template =  new RetryTemplate();
			template.setThrowLastExceptionOnExhausted(true);
			return template;
		}
        }
```
According to my understanding can be amended as ：
```
	@Configuration
	@ConditionalOnClass(RetryTemplate.class)
	public static class RetryAutoConfiguration {
		@Bean
                @ConditionalOnMissingBean
		public RetryTemplate retryTemplate() {
			RetryTemplate template =  new RetryTemplate();
			template.setThrowLastExceptionOnExhausted(true);
			return template;
		}
       }
```
If this can solve your problem? best wishes！

**Comment by iNikem** [comment]
Theoretically this is exactly what I would expect, yes :)

**Comment by holy12345** [comment]
@iNikem Thank you for your reply, if yes I will submit a PR to fix this issue. best wishes

**Comment by holy12345** [comment]
@ryanjbaxter Hello, do you think this change can be? If this change can solve this problem, I would like to submit a PR to be repaired.

**Comment by ryanjbaxter** [comment]
Sounds good to me.

**Comment by holy12345** [comment]
@iNikem I have submitted PR ：）

**Comment by dsyer** [comment]
Closed in error

**Comment by ryanjbaxter** [comment]
Fixed in #286
